# SOURCE MODEL (SM) TARGET (DEPENDENT) VARIABLE PRE-PROCESSING

## Importing Python Libraries

In [1]:
import numpy as np
import pandas as pd

print("Import Completed")

Import Completed


## Importing and Loading the Source Model (SM) Raw Data

In [2]:
source_model_raw_data = pd.read_csv("/Users/vidhyalakshmiparthasarathy/.CMVolumes/Google-Drive-pbvidhyauk6/QMUL_MSc_Project/credit_risk_scoring_project/data/transfer_learning_models_raw_data/source_model_raw_data.csv")

print("Data Import Completed")

/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3165: DtypeWarning: Columns (20) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


Data Import Completed


In [3]:
# Checking the size of the Source Model (SM) Raw Data

source_model_raw_data.shape

(524282, 152)

In [4]:
# Checking the first 3 column names of the dataset

source_model_raw_data.columns[[0, 1, 2]]

Index(['Unnamed: 0', 'id', 'member_id'], dtype='object')

In [5]:
# Removing the first index column from the dataset

source_model_raw_data.drop(source_model_raw_data.columns[0], axis = 1, inplace = True)

print("Execution Completed")

Execution Completed


In [6]:
# Re-checking the size of the Source Model (SM) Raw Data

source_model_raw_data.shape

(524282, 151)

In [7]:
# Displying the first 5 data instances of the Source Model (SM) Raw Data

source_model_raw_data.head()

id  member_id  loan_amnt  funded_amnt  funded_amnt_inv        term  \
0  136243871        NaN    15000.0      15000.0          15000.0   36 months   
1   63396131        NaN    15000.0      15000.0          15000.0   60 months   
2  130657021        NaN    15000.0      15000.0          15000.0   60 months   
3   60635254        NaN    25000.0      25000.0          25000.0   60 months   
4  129371880        NaN    36000.0      36000.0          36000.0   60 months   

   int_rate  installment grade sub_grade  ... debt_settlement_flag_date  \
0     10.08       484.58     B        B1  ...                       NaN   
1      8.18       305.44     B        B1  ...                       NaN   
2     11.98       333.52     B        B5  ...                       NaN   
3     15.61       602.79     D        D1  ...                       NaN   
4     21.85       991.22     D        D5  ...                       NaN   

  settlement_status settlement_date  settlement_amount settlement_percentage  \
0               NaN             NaN                NaN                   NaN   
1               NaN             NaN                NaN                   NaN   
2               NaN             NaN                NaN                   NaN   
3               NaN             NaN                NaN                   NaN   
4               NaN             NaN                NaN                   NaN   

  settlement_term fico_range_low fico_range_high last_fico_range_high  \
0             NaN          730.0           734.0                619.0   
1             NaN          725.0           729.0                754.0   
2             NaN          720.0           724.0                709.0   
3             NaN          720.0           724.0                644.0   
4             NaN          690.0           694.0                749.0   

  last_fico_range_low  
0               615.0  
1               750.0  
2               705.0  
3               640.0  
4               745.0  

[5 rows x 151 columns]

In [8]:
# Displying the last 5 data instances of the Source Model (SM) Raw Data

source_model_raw_data.tail()

id  member_id  loan_amnt  funded_amnt  funded_amnt_inv  \
524277   52729049        NaN    15000.0      15000.0          15000.0   
524278   56030757        NaN    14500.0      14500.0          14500.0   
524279  126378519        NaN    25000.0      25000.0          25000.0   
524280  134925617        NaN    10000.0      10000.0          10000.0   
524281  116948856        NaN    25000.0      25000.0          25000.0   

              term  int_rate  installment grade sub_grade  ...  \
524277   60 months     19.19       390.68     E        E3  ...   
524278   36 months      9.17       462.25     B        B2  ...   
524279   60 months     15.05       595.41     C        C4  ...   
524280   36 months      6.83       308.00     A        A3  ...   
524281   36 months     16.02       879.18     C        C5  ...   

       debt_settlement_flag_date settlement_status settlement_date  \
524277                       NaN               NaN             NaN   
524278                       NaN               NaN             NaN   
524279                       NaN               NaN             NaN   
524280                       NaN               NaN             NaN   
524281                       NaN               NaN             NaN   

        settlement_amount settlement_percentage settlement_term  \
524277                NaN                   NaN             NaN   
524278                NaN                   NaN             NaN   
524279                NaN                   NaN             NaN   
524280                NaN                   NaN             NaN   
524281                NaN                   NaN             NaN   

       fico_range_low fico_range_high last_fico_range_high last_fico_range_low  
524277          665.0           669.0                524.0               520.0  
524278          690.0           694.0                814.0               810.0  
524279          700.0           704.0                739.0               735.0  
524280          675.0           679.0                754.0               750.0  
524281          700.0           704.0                639.0               635.0  

[5 rows x 151 columns]

## Creating and Pre-Processing the new Target (Dependent) Variable - "fico_score_status"

### Creation of a new Target Variable Column named "fico_score_status" for the Source Model (SM)

### Creating a new column named "FICO Average Latest" by taking the average of the 2 attributes "last_fico_range_high" and "last_fico_range_low"

In [9]:
# Initially updating all the values of the "fico_average_latest" column with the average/mean of the latest/last fico score low and high values

fico_average_latest = (source_model_raw_data['last_fico_range_low'] + source_model_raw_data['last_fico_range_high'])/2
source_model_raw_data['fico_average_latest'] = fico_average_latest

print("Execution Completed")

Execution Completed


In [10]:
# Displying the first 5 data instances of the Source Model (SM) Data

source_model_raw_data.head()

id  member_id  loan_amnt  funded_amnt  funded_amnt_inv        term  \
0  136243871        NaN    15000.0      15000.0          15000.0   36 months   
1   63396131        NaN    15000.0      15000.0          15000.0   60 months   
2  130657021        NaN    15000.0      15000.0          15000.0   60 months   
3   60635254        NaN    25000.0      25000.0          25000.0   60 months   
4  129371880        NaN    36000.0      36000.0          36000.0   60 months   

   int_rate  installment grade sub_grade  ... settlement_status  \
0     10.08       484.58     B        B1  ...               NaN   
1      8.18       305.44     B        B1  ...               NaN   
2     11.98       333.52     B        B5  ...               NaN   
3     15.61       602.79     D        D1  ...               NaN   
4     21.85       991.22     D        D5  ...               NaN   

  settlement_date settlement_amount  settlement_percentage settlement_term  \
0             NaN               NaN                    NaN             NaN   
1             NaN               NaN                    NaN             NaN   
2             NaN               NaN                    NaN             NaN   
3             NaN               NaN                    NaN             NaN   
4             NaN               NaN                    NaN             NaN   

  fico_range_low fico_range_high last_fico_range_high last_fico_range_low  \
0          730.0           734.0                619.0               615.0   
1          725.0           729.0                754.0               750.0   
2          720.0           724.0                709.0               705.0   
3          720.0           724.0                644.0               640.0   
4          690.0           694.0                749.0               745.0   

  fico_average_latest  
0               617.0  
1               752.0  
2               707.0  
3               642.0  
4               747.0  

[5 rows x 152 columns]

In [11]:
# Conditional handling of zero values in "last_fico_range_low" column or "last_fico_range_high" column or in both the columns

#condition_1 = (source_model_raw_data['last_fico_range_low'] != 0.0) & (source_model_raw_data['last_fico_range_high'] == 0.0)
#condition_2 = (source_model_raw_data['last_fico_range_low'] == 0.0) & (source_model_raw_data['last_fico_range_high'] != 0.0)
#condition_3 = (source_model_raw_data['last_fico_range_low'] == 0.0) & (source_model_raw_data['last_fico_range_high'] == 0.0)
#condition_4 = (source_model_raw_data['last_fico_range_low'] != 0.0) & (source_model_raw_data['last_fico_range_high'] != 0.0)

# Set the value of "fico_average_latest" to the value in "last_fico_range_low"; if "last_fico_range_high" value is zero
source_model_raw_data['fico_average_latest'] = np.where((source_model_raw_data['last_fico_range_low'] != 0.0) & (source_model_raw_data['last_fico_range_high'] == 0.0), source_model_raw_data['last_fico_range_low'], source_model_raw_data['fico_average_latest'])

# Set the value of "fico_average_latest" to the value in "last_fico_range_high"; if "last_fico_range_low" value is zero
source_model_raw_data['fico_average_latest'] = np.where((source_model_raw_data['last_fico_range_low'] == 0.0) & (source_model_raw_data['last_fico_range_high'] != 0.0), source_model_raw_data['last_fico_range_high'], source_model_raw_data['fico_average_latest'])

# Last Observation Carried Forward (LOCF) Method
# Set the value of "fico_average_latest" to the mean value of "fico_range_low" and "fico_range_high"; if both "last_fico_range_low" and "last_fico_range_high" values are zero
source_model_raw_data['fico_average_latest'] = np.where((source_model_raw_data['last_fico_range_low'] == 0.0) & (source_model_raw_data['last_fico_range_high'] == 0.0), (source_model_raw_data['fico_range_low'] + source_model_raw_data['fico_range_high'])/2, source_model_raw_data['fico_average_latest'])

# Set the value of "fico_average_latest" to the mean value of "last_fico_range_low" and "last_fico_range_high"; if both "last_fico_range_low" and "last_fico_range_high" values are not equal to zero
source_model_raw_data['fico_average_latest'] = np.where((source_model_raw_data['last_fico_range_low'] != 0.0) & (source_model_raw_data['last_fico_range_high'] != 0.0), (source_model_raw_data['last_fico_range_low'] + source_model_raw_data['last_fico_range_high'])/2, source_model_raw_data['fico_average_latest'])


print("Execution Completed")

Execution Completed


In [12]:
# Checking the datatype of the newly added column "fico_average_latest"

source_model_raw_data['fico_average_latest'].dtype

dtype('float64')

In [13]:
# Checking the size of the newly added column "fico_average_latest"

source_model_raw_data['fico_average_latest'].shape

(524282,)

In [14]:
# Checking the descriptive statistics of the column "fico_average_latest"

source_model_raw_data['fico_average_latest'].describe()

count    524282.000000
mean        684.879289
std          72.908781
min         499.000000
25%         652.000000
50%         697.000000
75%         732.000000
max         847.500000
Name: fico_average_latest, dtype: float64

In [15]:
# Checking the minimum fico average score value

source_model_raw_data['fico_average_latest'].min()

499.0

In [16]:
# Checking the maximum fico average score value

source_model_raw_data['fico_average_latest'].max()

847.5

In [17]:
# Method 1 - Querying and verifying the list of rows with missing values in "fico_average_latest"

source_model_raw_data.query("fico_average_latest == 0.0")

Empty DataFrame
Columns: [id, member_id, loan_amnt, funded_amnt, funded_amnt_inv, term, int_rate, installment, grade, sub_grade, emp_title, emp_length, home_ownership, annual_inc, verification_status, issue_d, loan_status, pymnt_plan, url, desc, purpose, title, zip_code, addr_state, dti, delinq_2yrs, earliest_cr_line, inq_last_6mths, mths_since_last_delinq, mths_since_last_record, open_acc, pub_rec, revol_bal, revol_util, total_acc, initial_list_status, out_prncp, out_prncp_inv, total_pymnt, total_pymnt_inv, total_rec_prncp, total_rec_int, total_rec_late_fee, recoveries, collection_recovery_fee, last_pymnt_d, last_pymnt_amnt, next_pymnt_d, last_credit_pull_d, collections_12_mths_ex_med, mths_since_last_major_derog, policy_code, application_type, annual_inc_joint, dti_joint, verification_status_joint, acc_now_delinq, tot_coll_amt, tot_cur_bal, open_acc_6m, open_il_6m, open_il_12m, open_il_24m, mths_since_rcnt_il, total_bal_il, il_util, open_rv_12m, open_rv_24m, max_bal_bc, all_util, total_rev_hi_lim, inq_fi, total_cu_tl, inq_last_12m, acc_open_past_24mths, avg_cur_bal, bc_open_to_buy, bc_util, chargeoff_within_12_mths, delinq_amnt, mo_sin_old_il_acct, mo_sin_old_rev_tl_op, mo_sin_rcnt_rev_tl_op, mo_sin_rcnt_tl, mort_acc, mths_since_recent_bc, mths_since_recent_bc_dlq, mths_since_recent_inq, mths_since_recent_revol_delinq, num_accts_ever_120_pd, num_actv_bc_tl, num_actv_rev_tl, num_bc_sats, num_bc_tl, num_il_tl, num_op_rev_tl, num_rev_accts, num_rev_tl_bal_gt_0, num_sats, num_tl_120dpd_2m, ...]
Index: []

[0 rows x 152 columns]

In [18]:
# Method 2 - Verifying the list of rows with "NaN" values in "fico_average_latest"

source_model_raw_data['fico_average_latest'].isnull().sum()

0

In [19]:
# Displaying the rows with null values for the column "fico_average_latest"

fico_null_rows = pd.isnull(source_model_raw_data['fico_average_latest'])
source_model_raw_data[fico_null_rows]

Empty DataFrame
Columns: [id, member_id, loan_amnt, funded_amnt, funded_amnt_inv, term, int_rate, installment, grade, sub_grade, emp_title, emp_length, home_ownership, annual_inc, verification_status, issue_d, loan_status, pymnt_plan, url, desc, purpose, title, zip_code, addr_state, dti, delinq_2yrs, earliest_cr_line, inq_last_6mths, mths_since_last_delinq, mths_since_last_record, open_acc, pub_rec, revol_bal, revol_util, total_acc, initial_list_status, out_prncp, out_prncp_inv, total_pymnt, total_pymnt_inv, total_rec_prncp, total_rec_int, total_rec_late_fee, recoveries, collection_recovery_fee, last_pymnt_d, last_pymnt_amnt, next_pymnt_d, last_credit_pull_d, collections_12_mths_ex_med, mths_since_last_major_derog, policy_code, application_type, annual_inc_joint, dti_joint, verification_status_joint, acc_now_delinq, tot_coll_amt, tot_cur_bal, open_acc_6m, open_il_6m, open_il_12m, open_il_24m, mths_since_rcnt_il, total_bal_il, il_util, open_rv_12m, open_rv_24m, max_bal_bc, all_util, total_rev_hi_lim, inq_fi, total_cu_tl, inq_last_12m, acc_open_past_24mths, avg_cur_bal, bc_open_to_buy, bc_util, chargeoff_within_12_mths, delinq_amnt, mo_sin_old_il_acct, mo_sin_old_rev_tl_op, mo_sin_rcnt_rev_tl_op, mo_sin_rcnt_tl, mort_acc, mths_since_recent_bc, mths_since_recent_bc_dlq, mths_since_recent_inq, mths_since_recent_revol_delinq, num_accts_ever_120_pd, num_actv_bc_tl, num_actv_rev_tl, num_bc_sats, num_bc_tl, num_il_tl, num_op_rev_tl, num_rev_accts, num_rev_tl_bal_gt_0, num_sats, num_tl_120dpd_2m, ...]
Index: []

[0 rows x 152 columns]

In [20]:
# Querying and verifying the list of rows with "fico_average_latest" score values less than the expected/actual minimum score of 300 as per FICO Score Standards

source_model_raw_data.query("fico_average_latest < 300")

Empty DataFrame
Columns: [id, member_id, loan_amnt, funded_amnt, funded_amnt_inv, term, int_rate, installment, grade, sub_grade, emp_title, emp_length, home_ownership, annual_inc, verification_status, issue_d, loan_status, pymnt_plan, url, desc, purpose, title, zip_code, addr_state, dti, delinq_2yrs, earliest_cr_line, inq_last_6mths, mths_since_last_delinq, mths_since_last_record, open_acc, pub_rec, revol_bal, revol_util, total_acc, initial_list_status, out_prncp, out_prncp_inv, total_pymnt, total_pymnt_inv, total_rec_prncp, total_rec_int, total_rec_late_fee, recoveries, collection_recovery_fee, last_pymnt_d, last_pymnt_amnt, next_pymnt_d, last_credit_pull_d, collections_12_mths_ex_med, mths_since_last_major_derog, policy_code, application_type, annual_inc_joint, dti_joint, verification_status_joint, acc_now_delinq, tot_coll_amt, tot_cur_bal, open_acc_6m, open_il_6m, open_il_12m, open_il_24m, mths_since_rcnt_il, total_bal_il, il_util, open_rv_12m, open_rv_24m, max_bal_bc, all_util, total_rev_hi_lim, inq_fi, total_cu_tl, inq_last_12m, acc_open_past_24mths, avg_cur_bal, bc_open_to_buy, bc_util, chargeoff_within_12_mths, delinq_amnt, mo_sin_old_il_acct, mo_sin_old_rev_tl_op, mo_sin_rcnt_rev_tl_op, mo_sin_rcnt_tl, mort_acc, mths_since_recent_bc, mths_since_recent_bc_dlq, mths_since_recent_inq, mths_since_recent_revol_delinq, num_accts_ever_120_pd, num_actv_bc_tl, num_actv_rev_tl, num_bc_sats, num_bc_tl, num_il_tl, num_op_rev_tl, num_rev_accts, num_rev_tl_bal_gt_0, num_sats, num_tl_120dpd_2m, ...]
Index: []

[0 rows x 152 columns]

In [21]:
# Querying and verifying the list of rows with "fico_average_latest" score values greater than the expected/actual maximum score of 850 as per FICO Score Standards

source_model_raw_data.query("fico_average_latest > 850")

Empty DataFrame
Columns: [id, member_id, loan_amnt, funded_amnt, funded_amnt_inv, term, int_rate, installment, grade, sub_grade, emp_title, emp_length, home_ownership, annual_inc, verification_status, issue_d, loan_status, pymnt_plan, url, desc, purpose, title, zip_code, addr_state, dti, delinq_2yrs, earliest_cr_line, inq_last_6mths, mths_since_last_delinq, mths_since_last_record, open_acc, pub_rec, revol_bal, revol_util, total_acc, initial_list_status, out_prncp, out_prncp_inv, total_pymnt, total_pymnt_inv, total_rec_prncp, total_rec_int, total_rec_late_fee, recoveries, collection_recovery_fee, last_pymnt_d, last_pymnt_amnt, next_pymnt_d, last_credit_pull_d, collections_12_mths_ex_med, mths_since_last_major_derog, policy_code, application_type, annual_inc_joint, dti_joint, verification_status_joint, acc_now_delinq, tot_coll_amt, tot_cur_bal, open_acc_6m, open_il_6m, open_il_12m, open_il_24m, mths_since_rcnt_il, total_bal_il, il_util, open_rv_12m, open_rv_24m, max_bal_bc, all_util, total_rev_hi_lim, inq_fi, total_cu_tl, inq_last_12m, acc_open_past_24mths, avg_cur_bal, bc_open_to_buy, bc_util, chargeoff_within_12_mths, delinq_amnt, mo_sin_old_il_acct, mo_sin_old_rev_tl_op, mo_sin_rcnt_rev_tl_op, mo_sin_rcnt_tl, mort_acc, mths_since_recent_bc, mths_since_recent_bc_dlq, mths_since_recent_inq, mths_since_recent_revol_delinq, num_accts_ever_120_pd, num_actv_bc_tl, num_actv_rev_tl, num_bc_sats, num_bc_tl, num_il_tl, num_op_rev_tl, num_rev_accts, num_rev_tl_bal_gt_0, num_sats, num_tl_120dpd_2m, ...]
Index: []

[0 rows x 152 columns]

In [22]:
# Converting the datatype of the column "fico_average_latest" from float64 to int64 to represent the fico scores

source_model_raw_data['fico_average_latest'] = source_model_raw_data['fico_average_latest'].astype(int)

print("Execution Completed")

Execution Completed


In [23]:
# Checking the modified datatype of the column "fico_average_latest"

source_model_raw_data['fico_average_latest'].dtype

dtype('int64')

In [24]:
# Re-checking the descriptive statistics of the column "fico_average_latest" after converting the datatype to int64

source_model_raw_data['fico_average_latest'].describe()

count    524282.000000
mean        684.879189
std          72.908558
min         499.000000
25%         652.000000
50%         697.000000
75%         732.000000
max         847.000000
Name: fico_average_latest, dtype: float64

In [25]:
# Displying the first 5 data instances of the Source Model (SM) Raw Data

source_model_raw_data.head()

id  member_id  loan_amnt  funded_amnt  funded_amnt_inv        term  \
0  136243871        NaN    15000.0      15000.0          15000.0   36 months   
1   63396131        NaN    15000.0      15000.0          15000.0   60 months   
2  130657021        NaN    15000.0      15000.0          15000.0   60 months   
3   60635254        NaN    25000.0      25000.0          25000.0   60 months   
4  129371880        NaN    36000.0      36000.0          36000.0   60 months   

   int_rate  installment grade sub_grade  ... settlement_status  \
0     10.08       484.58     B        B1  ...               NaN   
1      8.18       305.44     B        B1  ...               NaN   
2     11.98       333.52     B        B5  ...               NaN   
3     15.61       602.79     D        D1  ...               NaN   
4     21.85       991.22     D        D5  ...               NaN   

  settlement_date settlement_amount  settlement_percentage settlement_term  \
0             NaN               NaN                    NaN             NaN   
1             NaN               NaN                    NaN             NaN   
2             NaN               NaN                    NaN             NaN   
3             NaN               NaN                    NaN             NaN   
4             NaN               NaN                    NaN             NaN   

  fico_range_low fico_range_high last_fico_range_high last_fico_range_low  \
0          730.0           734.0                619.0               615.0   
1          725.0           729.0                754.0               750.0   
2          720.0           724.0                709.0               705.0   
3          720.0           724.0                644.0               640.0   
4          690.0           694.0                749.0               745.0   

  fico_average_latest  
0                 617  
1                 752  
2                 707  
3                 642  
4                 747  

[5 rows x 152 columns]

In [26]:
# Displying the last 5 data instances of the Source Model (SM) Raw Data

source_model_raw_data.tail()

id  member_id  loan_amnt  funded_amnt  funded_amnt_inv  \
524277   52729049        NaN    15000.0      15000.0          15000.0   
524278   56030757        NaN    14500.0      14500.0          14500.0   
524279  126378519        NaN    25000.0      25000.0          25000.0   
524280  134925617        NaN    10000.0      10000.0          10000.0   
524281  116948856        NaN    25000.0      25000.0          25000.0   

              term  int_rate  installment grade sub_grade  ...  \
524277   60 months     19.19       390.68     E        E3  ...   
524278   36 months      9.17       462.25     B        B2  ...   
524279   60 months     15.05       595.41     C        C4  ...   
524280   36 months      6.83       308.00     A        A3  ...   
524281   36 months     16.02       879.18     C        C5  ...   

       settlement_status settlement_date settlement_amount  \
524277               NaN             NaN               NaN   
524278               NaN             NaN               NaN   
524279               NaN             NaN               NaN   
524280               NaN             NaN               NaN   
524281               NaN             NaN               NaN   

        settlement_percentage settlement_term fico_range_low fico_range_high  \
524277                    NaN             NaN          665.0           669.0   
524278                    NaN             NaN          690.0           694.0   
524279                    NaN             NaN          700.0           704.0   
524280                    NaN             NaN          675.0           679.0   
524281                    NaN             NaN          700.0           704.0   

       last_fico_range_high last_fico_range_low fico_average_latest  
524277                524.0               520.0                 522  
524278                814.0               810.0                 812  
524279                739.0               735.0                 737  
524280                754.0               750.0                 752  
524281                639.0               635.0                 637  

[5 rows x 152 columns]

### Creating a new Target Variable Column named "fico_score_status"

> #### 1. FICO Scores between 300 to 669 ----> Represents Good- ------> 0
> #### 2. FICO Scores between 670 to 850 ----> Represents Good+ ------> 1

### Above range of FICO Scores are considered based on the information available in Experian Credit Bureau Official Website.

### As per the Experian Credit Bureau; a good FICO score is in the range of 670 to 739. Hence we classified any FICO score less than 670 as the class "Good-" and any FICO score equal to and above 670 as class "Good+"; in order to address the binary classification problem for the Source Model (SM).

In [27]:
# Creating a new column to track the "Good-" and "Good+" class values based on the fico scores in the column "fico_average_latest"

source_model_raw_data['fico_score_status_good-_good+'] = np.where(source_model_raw_data['fico_average_latest']<=669, "Good-", "Good+")

print("Execution Completed")

Execution Completed


In [31]:
# Verifying the total count of "Good-" and "Good+"" classes

source_model_raw_data['fico_score_status_good-_good+'].value_counts().sum()

524282

In [32]:
# Verifying the individual count of "Good-" and "Good+"" classes

source_model_raw_data['fico_score_status_good-_good+'].value_counts()

Good+    353585
Good-    170697
Name: fico_score_status_good-_good+, dtype: int64

In [33]:
# Displying the first 5 data instances after adding the new column "fico_score_status_good-_good+"

source_model_raw_data.head()

id  member_id  loan_amnt  funded_amnt  funded_amnt_inv        term  \
0  136243871        NaN    15000.0      15000.0          15000.0   36 months   
1   63396131        NaN    15000.0      15000.0          15000.0   60 months   
2  130657021        NaN    15000.0      15000.0          15000.0   60 months   
3   60635254        NaN    25000.0      25000.0          25000.0   60 months   
4  129371880        NaN    36000.0      36000.0          36000.0   60 months   

   int_rate  installment grade sub_grade  ... settlement_date  \
0     10.08       484.58     B        B1  ...             NaN   
1      8.18       305.44     B        B1  ...             NaN   
2     11.98       333.52     B        B5  ...             NaN   
3     15.61       602.79     D        D1  ...             NaN   
4     21.85       991.22     D        D5  ...             NaN   

  settlement_amount settlement_percentage  settlement_term fico_range_low  \
0               NaN                   NaN              NaN          730.0   
1               NaN                   NaN              NaN          725.0   
2               NaN                   NaN              NaN          720.0   
3               NaN                   NaN              NaN          720.0   
4               NaN                   NaN              NaN          690.0   

  fico_range_high last_fico_range_high last_fico_range_low  \
0           734.0                619.0               615.0   
1           729.0                754.0               750.0   
2           724.0                709.0               705.0   
3           724.0                644.0               640.0   
4           694.0                749.0               745.0   

  fico_average_latest fico_score_status_good-_good+  
0                 617                         Good-  
1                 752                         Good+  
2                 707                         Good+  
3                 642                         Good-  
4                 747                         Good+  

[5 rows x 153 columns]

In [34]:
# Creating a new column to track the binary values of the fico scores classes as 0 for "Good-" class and 1 for "Good+" class

source_model_raw_data['fico_score_status'] = np.where(source_model_raw_data['fico_score_status_good-_good+'] == "Good-", 0, 1)

print("Execution Completed")

Execution Completed


In [35]:
# Checking the size of the dataset after the inclusion of new columns

source_model_raw_data.shape

(524282, 154)

In [36]:
# Verifying the total count of the classes 0 and 1

source_model_raw_data['fico_score_status'].value_counts().sum()

524282

In [37]:
# Verifying the individual count of the classes 0 and 1

source_model_raw_data['fico_score_status'].value_counts()

1    353585
0    170697
Name: fico_score_status, dtype: int64

In [38]:
# Displying the first 5 data instances after adding the new column "fico_score_status"

source_model_raw_data.head()

id  member_id  loan_amnt  funded_amnt  funded_amnt_inv        term  \
0  136243871        NaN    15000.0      15000.0          15000.0   36 months   
1   63396131        NaN    15000.0      15000.0          15000.0   60 months   
2  130657021        NaN    15000.0      15000.0          15000.0   60 months   
3   60635254        NaN    25000.0      25000.0          25000.0   60 months   
4  129371880        NaN    36000.0      36000.0          36000.0   60 months   

   int_rate  installment grade sub_grade  ... settlement_amount  \
0     10.08       484.58     B        B1  ...               NaN   
1      8.18       305.44     B        B1  ...               NaN   
2     11.98       333.52     B        B5  ...               NaN   
3     15.61       602.79     D        D1  ...               NaN   
4     21.85       991.22     D        D5  ...               NaN   

  settlement_percentage settlement_term  fico_range_low fico_range_high  \
0                   NaN             NaN           730.0           734.0   
1                   NaN             NaN           725.0           729.0   
2                   NaN             NaN           720.0           724.0   
3                   NaN             NaN           720.0           724.0   
4                   NaN             NaN           690.0           694.0   

  last_fico_range_high last_fico_range_low fico_average_latest  \
0                619.0               615.0                 617   
1                754.0               750.0                 752   
2                709.0               705.0                 707   
3                644.0               640.0                 642   
4                749.0               745.0                 747   

  fico_score_status_good-_good+ fico_score_status  
0                         Good-                 0  
1                         Good+                 1  
2                         Good+                 1  
3                         Good-                 0  
4                         Good+                 1  

[5 rows x 154 columns]

In [39]:
# Displying the last 5 data instances after adding the new column "fico_score_status"

source_model_raw_data.tail()

id  member_id  loan_amnt  funded_amnt  funded_amnt_inv  \
524277   52729049        NaN    15000.0      15000.0          15000.0   
524278   56030757        NaN    14500.0      14500.0          14500.0   
524279  126378519        NaN    25000.0      25000.0          25000.0   
524280  134925617        NaN    10000.0      10000.0          10000.0   
524281  116948856        NaN    25000.0      25000.0          25000.0   

              term  int_rate  installment grade sub_grade  ...  \
524277   60 months     19.19       390.68     E        E3  ...   
524278   36 months      9.17       462.25     B        B2  ...   
524279   60 months     15.05       595.41     C        C4  ...   
524280   36 months      6.83       308.00     A        A3  ...   
524281   36 months     16.02       879.18     C        C5  ...   

       settlement_amount settlement_percentage settlement_term  \
524277               NaN                   NaN             NaN   
524278               NaN                   NaN             NaN   
524279               NaN                   NaN             NaN   
524280               NaN                   NaN             NaN   
524281               NaN                   NaN             NaN   

        fico_range_low fico_range_high last_fico_range_high  \
524277           665.0           669.0                524.0   
524278           690.0           694.0                814.0   
524279           700.0           704.0                739.0   
524280           675.0           679.0                754.0   
524281           700.0           704.0                639.0   

       last_fico_range_low fico_average_latest fico_score_status_good-_good+  \
524277               520.0                 522                         Good-   
524278               810.0                 812                         Good+   
524279               735.0                 737                         Good+   
524280               750.0                 752                         Good+   
524281               635.0                 637                         Good-   

       fico_score_status  
524277                 0  
524278                 1  
524279                 1  
524280                 1  
524281                 0  

[5 rows x 154 columns]

## Exporting and saving the Source Model (SM) Raw Data with pre-processed target variable into a csv file

In [40]:
# Exporting and saving the Source Model (SM) Raw Data with pre-processed target variable "fico_score_status" into a csv file

source_model_raw_data.to_csv("/Users/vidhyalakshmiparthasarathy/.CMVolumes/Google-Drive-pbvidhyauk6/QMUL_MSc_Project/credit_risk_scoring_project/data/source_model_raw_data/source_model_pp_target_raw_data.csv")

print("Data Export Completed")

Data Export Completed
